### PR TITLE
Differentiate log levels between user and system exceptions

### PR DESCRIPTION
**Pull request by rsyncv**

### CODE REVIEW

File: metacat-main/src/main/java/com/netflix/metacat/main/api/ExceptionMapper.java [hunk 1]
```diff
@@ -66,18 +67,26 @@ public void handleMetacatException(
...
+            logErrorLevel = true;
             status = HttpStatus.NOT_IMPLEMENTED.value();
         } else if (e instanceof MetacatUserMetadataException) {
             // TODO: This makes no sense
             status = HttpStatus.SEE_OTHER.value();
         } else if (e instanceof MetacatTooManyRequestsException) {
+            logErrorLevel = true;
             status = HttpStatus.TOO_MANY_REQUESTS.value();
         } else if (e instanceof MetacatUnAuthorizedException) {
+            logErrorLevel = true;
```
Comment (swaranga-netflix) [inline metacat-main/src/main/java/com/netflix/metacat/main/api/ExceptionMapper.java:79]: Should this be a user exception?

File: metacat-main/src/main/java/com/netflix/metacat/main/api/ExceptionMapper.java [hunk 2]
```diff
@@ -66,18 +67,26 @@ public void handleMetacatException(
...
             status = HttpStatus.FORBIDDEN.value();
         } else {
+            logErrorLevel = true;
             status = HttpStatus.INTERNAL_SERVER_ERROR.value();
         }
-        log.error(e.getLocalizedMessage(), e);
+        if (logErrorLevel) {
+            log.error(e.getLocalizedMessage(), e);
+        } else {
+            log.warn(e.getLocalizedMessage(), e);
```
Comment (swaranga-netflix) [inline metacat-main/src/main/java/com/netflix/metacat/main/api/ExceptionMapper.java:88]: We can also probably omit the stack trace for this one.

File: metacat-main/src/main/java/com/netflix/metacat/main/api/ExceptionMapper.java [hunk 3]
```diff
@@ -66,18 +67,26 @@ public void handleMetacatException(
         } else if (e instanceof MetacatNotFoundException) {
             status = HttpStatus.NOT_FOUND.value();
         } else if (e instanceof MetacatNotSupportedException) {
+            logErrorLevel = true;
             status = HttpStatus.NOT_IMPLEMENTED.value();
         } else if (e instanceof MetacatUserMetadataException) {
             // TODO: This makes no sense
             status = HttpStatus.SEE_OTHER.value();
         } else if (e instanceof MetacatTooManyRequestsException) {
+            logErrorLevel = true;
```
Comment (swaranga-netflix) [inline metacat-main/src/main/java/com/netflix/metacat/main/api/ExceptionMapper.java:76]: Also probably a user exception?

File: metacat-main/src/main/java/com/netflix/metacat/main/api/ExceptionMapper.java (hunk 2 above)
Comment (rsyncv) [reply metacat-main/src/main/java/com/netflix/metacat/main/api/ExceptionMapper.java:88]: Would like to keep logging stack trace for now, this PR is mainly focused on separating the two types so as to be more easily searchable and distinguishable in something like radar.

File: metacat-main/src/main/java/com/netflix/metacat/main/api/ExceptionMapper.java (hunk 1 above)
Comment (rsyncv) [reply metacat-main/src/main/java/com/netflix/metacat/main/api/ExceptionMapper.java:79]: Ok.

**Review by swaranga-netflix — APPROVED** [review]

**Pull request merged**